frontend/public/components/create-resource.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { match as RouterMatch } from 'react-router';
+import { Extension, useExtensions } from '@console/plugin-sdk';
+import {
+  CreateResource as CreateResourceExtension,
+  isCreateResource,
+} from '@console/dynamic-plugin-sdk';
+import { AsyncComponent } from './utils';
+import { referenceForExtensionModel } from '../module/k8s';
+
+type CreateYAMLPageProps = {
+  match: RouterMatch<{ plural: string; ns?: string }>;
+};
+
+const CreateYAMLPage: React.FC<CreateYAMLPageProps> = ({ match }) => (
+  <AsyncComponent
+    loader={() =>
+      import('./create-yaml' /* webpackChunkName: "create-yaml" */).then((m) => m.CreateYAML)
+    }
+    match={match}
+  />
+);
+
+type CreateResourceProps = CreateYAMLPageProps & {
+  DefaultPage?: React.ComponentType<CreateYAMLPageProps>;
+};
+
+const CreateResource: React.FC<CreateResourceProps> = ({ match, DefaultPage = CreateYAMLPage }) => {
+  const createResourceTypeGuard = React.useCallback(
+    (e: Extension): e is CreateResourceExtension => {
+      return (
+        isCreateResource(e) &&
+        referenceForExtensionModel(e.properties.model) === match.params.plural
+      );
+    },
+    [match.params.plural],
+  );
+  const extensionPages = useExtensions<CreateResourceExtension>(createResourceTypeGuard);
```

Comment:
We can factor out this logic into a hook that returns a single create resource extension given the `match.params.plural` arg. That way it can be reused here and in CreateOperandPage.